app/models/course/settings/lesson_plan_items.rb

```diff
@@ -0,0 +1,73 @@
+# frozen_string_literal: true
+#
+# This model facilitates displaying and setting of lesson plan item settings.
+#
+# To add lesson plan item settings to a course component, ensure that these two methods
+# are defined on the component's setting model
+# (see {Course::ControllerComponentHost::Settings::ClassMethods#settings_class}):
+#
+# - `#lesson_plan_item_settings` - see {#lesson_plan_item_settings} for details
+# - `#update_lesson_plan_item_setting` - see {#update} for details
+#
+# Lesson Plan Item settings are stored with the individual course components as all such items
+# e.g. Surveys and Videos, act as lesson plan items.
+#
+class Course::Settings::LessonPlanItems
```

Comment:
This class is now mostly used to persist the settings class, but I think we should load it in the `lesson_plan_item` controller (with a `@settings` instance variable), and provide methods to be used. A few things will need to change: 
 - This settings should also inherit from `Course::Settings::Component`. It will provide the functionality to check out enabled and disabled components. 
 - The [code introduced](https://github.com/coursemology-collab/coursemology2/blob/d985b5d099baa83be131da4ef671e7ea97d4a164/app/controllers/course/lesson_plan/items_controller.rb#L71-L100) in the `lesson_plan_item` controller should also reside within this class, allowing this class to manage the complexity. 
 - The hard part is mapping the component (Where we define the `lesson_plan_actable_class` and the settings class. I think once you inherit `Course::Settings::Component`, use the method `settings_class` on `enabled_components`. I think with this you should be able to properly construct the actable hash.

Reply:
> This class is now mostly used to persist the settings class, but I think we should load it in the lesson_plan_item controller (with a @settings instance variable), and provide methods to be used.

I agree that we should load it in the `lesson_plan_item` controller, but perhaps not as `@settings`.

Perhaps it would be helpful to agree on how to think about the settings framework. The initial settings classes were geared towards settings within a component. These are end-of-the-line settings components that directly get/set settings values within a sub-tree of settings that is confined only to the component. Each controller for a component gets its settings automatically loaded, if it exists.

Now we have a second type of settings class -- pan-component coordinator/consolidator settings components like notifications that don't actually get/set values directly. Instead, they call methods on the end-of-the-line settings components to do it.

`@settings` will be defined automatically for the `lesson_plan_item` controller once I define `Course::Settings::LessonPlanComponent` (for the LP collapsing settings).

How the setting auto-loading currently works:
1. Let the controller inherit from `Course::ComponentController` and define `#component` to associate the controller with the component.
1. Then, a `before_action` will load `component.settings` as `@settings`  automatically when the controller is loaded (see `Course::ComponentController`).
1. `component.settings` is an instance of the component's settings class. By default, the settings class name is inferred from the component name, e.g. the settings class for `Course::VideosComponent` is `Course::Settings::VideosComponent`. This can be over-ridden.

Since `Course::LessonPlan::ItemsController < Course::LessonPlan::Controller`, `@settings` will be defined automatically later. As per my earlier comment, perhaps we can let `@settings` be the end-of-the-line LP settings and `@items_settings` be the pan-component LP items settings?

This is already the case for `LessonPlanSettingsController`. Shall we do the same for `Course::LessonPlan::ItemsController`?

> This settings should also inherit from `Course::Settings::Component`. It will provide the functionality to check out enabled and disabled components.

`Course::Settings::Component` is meant for end-of-the-line settings components. It defines getter/setter helpers + loads the sub-tree of settings for a particular component. All other method calls are delegated.

I think inheriting from `Course::Settings::PanComponents` (#2731) works.

> The code introduced in the `lesson_plan_item` controller should also reside within this class, allowing this class to manage the complexity.

Yup, agreed. `Course::LessonPlan::ItemsController` should use `Course::Settings::LessonPlanItems` do the consolidation of settings for different components, instead of accessing directly for any individual component.